src/lib/Microsoft.Health.Fhir.Ingest/Data/MeasurementToEventMessageAsyncCollector.cs

```diff
@@ -39,6 +50,61 @@ public async Task AddAsync(IMeasurement item, CancellationToken cancellationToke
             await _eventHubService.SendAsync(eventList, partitionKey, cancellationToken).ConfigureAwait(false);
         }
 
+        public async Task AddAsync(IEnumerable<IMeasurement> items, CancellationToken cancellationToken = default)
+        {
+            EnsureArg.IsNotNull(items, nameof(items));
+
+            using (var hasher = _hashCodeFactory.CreateDeterministicHashCodeGenerator())
+            {
+                var submissionTasks = items
+                .GroupBy(m =>
+                {
+                    // cast as byte to restrict to 256 possible values. This will lead to a greater change of measurements ending up in the same bucket,
+                    // while providing partition keys with enough entropy for EventHub to better distribute them across partitions.
+                    var partitionKey = (byte)hasher.GenerateHashCode(m.DeviceId.ToLower());
```

Comment:
Here is where we restrict the hashcode to the values within the range of a `byte`

Reply:
I don't think we should use a cast here.  I just tested, if the byte value is < 0 it will become zero and if it is > 255 it will become 255 with this method.  It means we will be artifically sending most hashes to partition key 0 or 255.

I would suggest one of 3 things:
* Update Hash interface to a byte
* Update the Hash interface to a generic and this class uses  the of byte one
* Or update the hash interface to return string to match what event hub wants for a partition key.  This would have the ultimate flexibility, allowing us to freely switch between the old int based version, the new byte based version, or another yet to be determined method.

I prefer the last one personally.

Reply:
After some more digging it looks like there is no + operand defined for bytes which makes things tricker.  Adding two bytes together results in an integer.

To avoid use biasing towards the low or high end (0 or 255) I think we would need to do %256 and take the absolute value based on your current hasing algo

Reply:
Something like this should do the trick if you go with the 3rd option, replace return h with return  ((byte)(Math.Abs(h) % 256)).ToString();